3rdparty/libunwind/CMakeLists.txt

```diff
@@ -1,110 +1,153 @@
-    # Note that the C compiler is used for ASM as well.
-    CC=${CMAKE_C_COMPILER}
-    CXX=${CMAKE_CXX_COMPILER}
-    )
-
-string(TOUPPER "${CMAKE_BUILD_TYPE}" BUILD_TYPE)
-
-if ((BUILD_TYPE STREQUAL "DEBUG") OR (BUILD_TYPE STREQUAL "RELWITHDEBINFO"))
-    list(APPEND OPTS "--enable-debug=yes" "--enable-debug-frame==yes")
-    set(CFLAGS "${CFLAGS} -g")
-else()
-    list (APPEND OPTS "--enable-debug=no" "--enable-debug-frame==no")
-endif()
-
-include (ExternalProject)
-
-ExternalProject_Add(oelibcxx_unwind
-
-    # Copy libunwind source tree to build directory.
-    DOWNLOAD_COMMAND ${CMAKE_COMMAND} -E copy_directory
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind 
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind
-
-    # Remove -Wall as this is applied too late and will override -Wno-* flags in Clang.
-    # Instead we apply it ourselves using CFLAGS (see above).
-    COMMAND sed -e "s/-Wall//g"
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind/configure.ac >
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/configure.ac
-
-    # Generate autoconf scripts and run configure.
-    CONFIGURE_COMMAND cd ${CMAKE_CURRENT_BINARY_DIR}/libunwind && 
-        ./autogen.sh "${OPTS}" "CFLAGS=${CFLAGS}"
-
-    # Copy libunwind-common.h to libunwind-common.inc.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/include/libunwind-common.h
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/include/libunwind-common.inc
-
-    # Copy libunwind-common.h to the build directory (this file includes
-    # libunwind-common.inc).
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind-common.h
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/include/libunwind-common.h
-
-    # Copy Gstep.c to Gstep.inc.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind/src/x86_64/Gstep.c
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gstep.inc
-
-    # Copy Gstep.c to the the build directory.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/Gstep.c
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gstep.c
-
-    # Copy Gtrace.c to Gtrace.inc while removing the line containing
-    # "static __thread".
-    COMMAND grep -v "^static __thread"
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind/src/x86_64/Gtrace.c >
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gtrace.inc
-
-    # Copy Gtrace.c to the the build directory.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/Gtrace.c
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gtrace.c
-
-    # Copy setcontext.S over the one in the build directory.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/setcontext.S
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/setcontext.S
-
-    # Build libunwind.
-    BUILD_COMMAND $(MAKE) -C ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src
-
-    # Empty install command.
-    INSTALL_COMMAND "")
+configure_file(
```

Comment:
For next iteration:
One thing I noticed in my Windows build experiments was that even if I edit a test enclave's CMakeLists.txt, it triggered a rebuild of libunwind. I suspect it is due to something we are doing here.

Reply:
Good to know... I'll have to take a look.